[DOCS] Helm chart 2.0 docs

ian-at-airbyte opened the pull request:
## What

This PR adds documentation for Helm chart V2 in Airbyte's official docs. This includes two migration guides (Community and Enterprise), a new values.yaml reference doc, new documentation for generic OIDC, and updates existing values.yaml documentation to provide samples in both V1 and V2 form.

The general idea is that we provide both an end-to-end guide (cloud provider-neutral), and also maintain both V1 and V2 samples and discussions, until V2 is mandatory. At a point in the future, we can remove the V1 content once it's no longer officially supported.

**What's currently up in the air:**

I'm not exactly clear on the optimal migration path for Community users who don't have external storage, DB, and secrets management. We should determine what we want to happen here before publishing this.

## How

- Add two migration guides, one for Community and one for Enterprise. These include basic instructions to migrate customizations from V1 to V2, a list of all syntax changes from V1 to V2, and a map of each extraEnv environment variable to the V2 interface.
- Add values.yaml reference page + install plugin to display code samples directly from GitHub
- Add Generic OIDC setup docs to SSO docs
- Update existing deployment documentation. All values.yaml samples have customizatons in V1 and V2 form, separated with tabs. All tabs should be in sync so in long topics, you can just click the version you want and all tabs automatically switch.
- Update linter exceptions with some words to always accept.

I've tried to maintain a general standard for helm command samples that:

- Community uses an `airbyte` repo name and Enterprise features use an `airbyte-enterprise` repo name (this standard predates me, so I'm just continuing it)
- V1 samples use an `airbyte` namespace, V2 samples use an `airbyte-v2` namespace
- V2 samples specify a chart and airbyte version, and V1 samples do not

## Review guide

Changes will only be visible in the "Next" platform version (IE https://airbyte-docs-git-docs-helm-charts-20-airbyte-growth.vercel.app/platform/next/).

I've run this PR through Devin and asked it to verify both the values and helm commands. It caught a couple of mistakes but seemed otherwise satisfied based on its understanding of how the code works... we should not assume Devin or I are incapable of making a mistake, though.

There's no specific review order, but I think this would make the most sense:

**Migration guides**

These docs demonstrate the basic, expected process to move from V1 to V2. They don't cover every possible configuration, but they cover all the standard ones. They also provide a mapping of V1 environment variables and keys to those expected in V2.

- [Enterprise - platform/next/enterprise-setup/chart-v2-enterprise](https://airbyte-docs-git-docs-helm-charts-20-airbyte-growth.vercel.app/platform/next/enterprise-setup/chart-v2-enterprise)
- [Community - platform/next/deploying-airbyte/chart-v2-community](https://airbyte-docs-git-docs-helm-charts-20-airbyte-growth.vercel.app/platform/next/deploying-airbyte/chart-v2-community)

Both of these guides are very similar, but the Community one is stripped down because they don't have as many features.

**Values.yaml reference**

- [/platform/next/deploying-airbyte/values](https://airbyte-docs-git-docs-helm-charts-20-airbyte-growth.vercel.app/platform/next/deploying-airbyte/values) - Autogenerated samples directly from GitHub. Having a full reference doc might be preferable but these files are pretty helpful as it is.

**Instructions to set up generic OIDC**

A new feature of Helm V2, I've added setup instructions for generic OIDC setup in Okta and EntraID. I made slight updates to previously existing content to facilitate the new addition. However, I did not rewrite any of the existing OIDC and Cloud documentation as it's out of scope for this work.

- [platform/next/access-management/sso-providers/azure-entra-id](https://airbyte-docs-git-docs-helm-charts-20-airbyte-growth.vercel.app/platform/next/access-management/sso-providers/azure-entra-id)
- [/platform/next/access-management/sso-providers/okta](https://airbyte-docs-git-docs-helm-charts-20-airbyte-growth.vercel.app/platform/next/access-management/sso-providers/okta)

**Updated deployment content**

In existing deployment docs spread around the site, I identified files that had values.yaml samples and updated them. Now, they have two tabs: Helm chart V1 and Helm chart V2. Readers can choose the appropriate version for their needs. In a few cases, I did not provide tabs, because all samples on the page appeared the same in V1 and V2.

- [platform/next/deploying-airbyte/](https://airbyte-docs-git-docs-helm-charts-20-airbyte-growth.vercel.app/platform/next/deploying-airbyte/) + all child pages - All values.yaml samples now have a tab interface with the option to choose between V1 and V2 mode. In a few cases, I did not change anything because V1 and V2 were identical.
- [/platform/next/enterprise-setup/audit-logging](https://airbyte-docs-git-docs-helm-charts-20-airbyte-growth.vercel.app/platform/next/enterprise-setup/audit-logging) - Added V2 configuration.
- [/platform/next/enterprise-setup/scaling-airbyte](https://airbyte-docs-git-docs-helm-charts-20-airbyte-growth.vercel.app/platform/next/enterprise-setup/scaling-airbyte)
- Various other small updates, spread across numerous files, to add V2 samples. Where appropriate, I tried to clean up helm commands to try to keep the general look in sync between V1 and V2.
 
## User Impact

People can migrate to Helm chart V2.

## Can this PR be safely reverted and rolled back?

- [x] YES 💚
- [ ] NO ❌

ian-at-airbyte committed:
Add ability to import code samples from GitHub

ian-at-airbyte committed:
Merge branch 'master' into docs-helm-charts-2.0

ian-at-airbyte committed:
Added a first draft with some portions completed. Still quite rough and many examples are missing.

ian-at-airbyte committed:
Added database and external logging. Cleaned up some introductory content.

ian-at-airbyte committed:
Merge branch 'master' into docs-helm-charts-2.0

ian-at-airbyte committed:
SSO home page

ian-at-airbyte committed:
Added some new linting exceptions.

ian-at-airbyte committed:
Added Okta and EntraID generic OIDC info

ian-at-airbyte committed:
Standardizing on values.yaml instead of values.yml for illustrative purposes.

ian-at-airbyte committed:
Split migration guide into Community and Enterprise since they're somewhat different.

ian-at-airbyte committed:
Fix a broken link

ian-at-airbyte committed:
Added Helm chart V2 syntax

ian-at-airbyte committed:
Merge branch 'master' into docs-helm-charts-2.0

ian-at-airbyte committed:
Fixed an indentation issue

ian-at-airbyte committed:
Added Helm chart V2 to authentication docs

ian-at-airbyte committed:
Added a bit of context to the Configuring Airbyte page

ian-at-airbyte committed:
Rephrased some things.

ian-at-airbyte committed:
Added database config for Helm v2

ian-at-airbyte committed:
Add storage

ian-at-airbyte committed:
Updated deployment guide, secrets, audit logging

ian-at-airbyte committed:
Slight adjustments to service account upgrade topic, but nothing specific to chart V2

ian-at-airbyte committed:
Added upgrade command to Community upgrade

ian-at-airbyte committed:
Slight adjustment on collecting metrics. No substantive change to V2.

ian-at-airbyte committed:
Updated telemetry collection. Mostly copy edits. The interface appears the same.

ian-at-airbyte committed:
General cleanup and add Helm V2 to Upgrading Airbyte

ian-at-airbyte committed:
Added chart v2 to Helm chart V2

ian-at-airbyte committed:
Add a note that generic OIDC can only be used with Helm chart 2.0

ian-at-airbyte committed:
Repair an error in scaling Airbyte

ian-at-airbyte committed:
Minor adjustments to the telemetry topic to improve alignment

ian-at-airbyte committed:
Added environment variables -> Helm chart V2 reference table

ian-at-airbyte committed:
Update Enterprise implementation guide

ian-at-airbyte committed:
Fixed a few issues and clarified deployment commands based on Devin's feedback

ian-at-airbyte committed:
Added validation steps and cleaned up the helm upgrade command sample

ian-at-airbyte committed:
Format tables

ian-at-airbyte committed:
Prevent tables with long words from overflowing their parent container

ian-at-airbyte committed:
Slight clarifications in both migration guides

devin-ai-integration[bot] committed:
Remove columns 3 and 4 from environment variable mapping table

Keep only the before/after columns as requested, removing the empty column and description column to create a cleaner 2-column table format.

Co-Authored-By: ian.alton@airbyte.io <ian.alton@airbyte.io>

git-phu commented on docs/platform/deploying-airbyte/chart-v2-community.mdx:
````diff
@@ -0,0 +1,338 @@
+
+```yaml title="values.yaml using Helm chart V2"
+global:
+  jobs:
+    kube:
+      nodeSelector:
+        type: jobs
````
Something I encountered when migrating to the V2 charts:
```
  jobs:
    kube:
      nodeSelector:
        type: jobs
```
As it turns out, this corresponds to the `JOB_KUBE_NODE_SELECTORS` env var only, so we need to specify the other node selectors explicitly in the V2 chart as well.

```suggestion
  jobs:
    kube:
      nodeSelector:
        type: jobs
      scheduling:
        check:
          nodeSelectors:
            type: jobs
        discover:
          nodeSelectors:
            type: jobs
```

The chart is also currently missing a key for spec jobs so we should add that to the chart as well

ian-at-airbyte replied:
Thanks. This makes sense but wasn't what I was expecting either. I've added this to all 3 occurrences of this situation in the docs. I've also added a spec entry on the assumption we will add that to the chart at some point, but  I guess technically that won't work today. I'll report it to the team and see what they want to do.

git-phu commented on docs/platform/deploying-airbyte/chart-v2-community.mdx:
```diff
@@ -0,0 +1,338 @@
+
+### Configure external database, bucket storage, and secrets manager
+
+<!-- This might be a non-starter for community. Can we reasonably ask them to do this? -->
+
+<!-- Although Airbyte provides its own basic database, storage, and secrets management, most people opt to use their own. If you don't already do this, you must enable these before migrating.
```
Would it help to add some explanation for why we want to recommend setting up an external database, storage, and secrets manager?

The V2 chart does still support the basic embedded versions of these services like the V1 chart did, so I could potentially see how some users would not understand why this is listed as a step for upgrading the chart when it technically isn't required.

git-phu reviewed (approved):
Nice! Lgtm for the parts that were relevant for me when I was migrating to the V2 charts.

For context, my experience with chart migrations was to ensure that fresh installs of airbyte in our ci environment behaved the same way whether we were using the V1 or V2 charts, so I did not deal with migrating existing instances of airbyte deployed with V1 charts to using V2 charts.

So for my use case, the migration (for oss deploys) was quite straightforward.  The only thing I really had to change was how to specify node selectors for job pods and I left a comment about that on the PR.

ian-at-airbyte committed:
Updated node selector syntax to reflect that scheduling nodes are not automatically inherited as I expected. Thanks Peter.

Peter noticed it actually was mandatory to specify scheduling, discover, and spec nodeSelectors. Otherwise, they'll run in the static node group, which we don't want. So inheritance is only implied, but not actual. The specific node must be set for each. I'm also adding an entry for spec BUT this is currently missing from the chart. It needs to be added, and I'm documenting it here on the assumption that it will be added. However, technically, as of today, it doesn't work.

ian-at-airbyte committed:
Added a warning not to upgrade without setting up external resources.

ian-at-airbyte committed:
Comment out an unused section

perangel commented on docs/platform/deploying-airbyte/chart-v2-community.mdx:
```diff
@@ -0,0 +1,348 @@
+
+The following versions of Airbyte can use Helm chart V2:
+
+- Airbyte version 1.6.0 and later, if installed and managed with Helm _and_ connected to an external database, storage, and secrets manager
```
Not sure if this will confuse folks, but I believe that technically if you are using an external database and NOT using any external secrets manager, then the DB is where your secrets are stored, so they will effectively be persisted along with your other data. If folks are not currently using an external secrets manager, that's probably ok for the migration, as along as they are using an external database.

ian-at-airbyte replied:
Right, I forgot about this. Thanks!

perangel commented on docs/platform/deploying-airbyte/chart-v2-community.mdx:
```diff
@@ -0,0 +1,348 @@
+
+When moving to Helm chart V2, deploy Airbyte with a new namespace and use a fresh values and secrets file. It is possible to do a straight upgrade, but different Airbyte users have different and sometimes complex configurations that could produce unique and unexpected situations during the upgrade. By doing a fresh install, you create a separate environment that's easier to troubleshoot if someth ...
+
+### Add and index the Helm chart 2.0 repo
```
Let's be consistent with `V2` vs `2.0`. 
Also, there is no action to be performed: re indexing, so I would just not mention that
```suggestion
### Add V2 chart repo
```

perangel commented on docs/platform/deploying-airbyte/chart-v2-community.mdx:
```diff
@@ -0,0 +1,348 @@
+
+### Add and index the Helm chart 2.0 repo
+
+Helm chart V2 uses a separate repo than V1 did. In your command line tool, add this repo and index it.
```
Perhaps its worth visualizing this in a code block for clarity:
```
# v1 chart
airbyte         https://airbytehq.github.io/helm-charts
# v2 chart
airbyte-v2     https://airbytehq.github.io/charts

# NOTE: the repository URL has changed for v2
```

ian-at-airbyte commented on docs/platform/access-management/sso-providers/okta.md:
```diff
@@ -6,40 +6,41 @@ products: oss-enterprise, cloud-teams
+:::
 
-On the following screen you'll need to configure all parameters for your Okta application:
+You need to create a new Okta OIDC App Integration for your Airbyte. Documentation on how to do this in Okta can be found [here](https://help.okta.com/en-us/content/topics/apps/apps_app_integration_wizard_oidc.htm). You should create an app integration with **OIDC - OpenID Connect** as the sign-in method and **Web Application** as the application type:
```
Nope, just a typo :) Thanks.

ian-at-airbyte committed:
Update docs/platform/deploying-airbyte/chart-v2-community.mdx

Co-authored-by: perangel <perangel@gmail.com>

ian-at-airbyte committed:
Update docs/platform/deploying-airbyte/chart-v2-community.mdx

Co-authored-by: perangel <perangel@gmail.com>

ian-at-airbyte committed:
Applied some edits from Angel. Also set migration guides to draft, awaiting decisions here.

ian-at-airbyte committed:
Merge branch 'docs-helm-charts-2.0' of https://github.com/airbytehq/airbyte into docs-helm-charts-2.0

ian-at-airbyte committed:
Merge branch 'master' into docs-helm-charts-2.0

ian-at-airbyte committed:
Fixed a build error

ian-at-airbyte commented:
I've addressed all comments. Thanks everyone.

Currently the migration guide is set to "draft" so Alex/Bryce can figure out an optimal rollout path with customers and community. However, all values.yaml samples with the new V1/V2 tabs are ready to go.

ian-at-airbyte committed:
Merge branch 'master' into docs-helm-charts-2.0
